lib/index.js

```diff
@@ -9,17 +9,13 @@ exports.register = (server, options, next) => {
       return reply.continue();
     }
 
-    let rate = Object.assign({}, options.defaultRate(request));
     const routeSettings = request.route.settings.plugins.rateLimit;
-
-    if (routeSettings) {
-      if (routeSettings.enabled === false) {
-        return reply.continue();
-      }
-
-      rate = routeSettings.rate(request);
+    if (!routeSettings || !routeSettings.enabled) {
+      return reply.continue();
     }
 
+    const rate = routeSettings.rate ? routeSettings.rate(request) : options.defaultRate(request);
```

Comment:
Thoughts on doing `typeof routeSettings.rate === 'function' ? ...` to ensure that it's callable?